app/models/patient/triage_status.rb

```diff
@@ -73,19 +73,27 @@ def vaccination_history_requires_triage?
 
   private
 
+  def vaccinated?
+    VaccinatedCriteria.call(programme:, patient:, vaccination_records:)
```

Comment:
Since this method is being called multiple times I think it's worth caching the value.

```suggestion
    @vaccinated ||= VaccinatedCriteria.call(programme:, patient:, vaccination_records:)
```